renku_notebooks/api/amalthea_patches/general.py

```diff
@@ -201,32 +201,31 @@ def oidc_unverified_email(server: "UserServer"):
 
 def dev_shm(server: "UserServer"):
     patches = []
-    if server.server_options.storage:
-        patches.append(
-            {
-                "type": "application/json-patch+json",
-                "patch": [
-                    {
-                        "op": "add",
-                        "path": "/statefulset/spec/template/spec/volumes/-",
-                        "value": {
-                            "name": "shm",
-                            "emptyDir": {
-                                "medium": "Memory",
-                                # NOTE: We are giving /dev/shm up to half of the memory request
-                                "sizeLimit": int(server.server_options.storage / 2),
-                            },
+    patches.append(
+        {
+            "type": "application/json-patch+json",
+            "patch": [
+                {
+                    "op": "add",
+                    "path": "/statefulset/spec/template/spec/volumes/-",
+                    "value": {
+                        "name": "shm",
+                        "emptyDir": {
+                            "medium": "Memory",
+                            # NOTE: We are giving /dev/shm up to half of the memory request
+                            "sizeLimit": int(server.server_options.memory / 2),
```

Comment:
I am not sure that `server.server_options.memory` is always non null. So do a check and put a default if it is not set. 
```suggestion
                            "sizeLimit": int(server.server_options.memory / 2) if isinstance(server.server_options.memory, int) else "1Gi",
```